### PR TITLE
Add TAs to class list

### PR DESCRIPTION
**Pull request by Maryam676**
- Resolve merge conflicts from earlier branch
- This branch contains commits that were left out of ta-class from an earlier merge

### CODE REVIEW

File: src/test/java/com/google/sps/servlets/TATest.java [hunk 1]
```diff
@@ -50,24 +54,133 @@ public void tearDown() {
...
+    // Obtain the corresponding class entity and its TA list
+    Entity testEntity = datastore.get(classEntity.getKey());
+    List<String> listOfClassTAs = (List) testEntity.getProperty("taList");
+
+    // Verify the new TA was added to the class TA roster
+    assertTrue(listOfClassTAs.contains("test@google.com"));
+  }
+
+  @Test
+  // Add one TA to an empty class TA list
```
Comment (mdchu) [inline src/test/java/com/google/sps/servlets/TATest.java:90]: I think this comment needs updating, same comment on multiple tests in this file

File: src/main/java/com/google/sps/servlets/AddTA.java [hunk 2]
```diff
@@ -18,8 +22,7 @@
 // Once class owner submits a TA email, retrieve that user and add them as a TA to the class
 @WebServlet("/add-ta")
 public class AddTA extends HttpServlet {
-  private FirebaseAuth authInstance;
-
+  FirebaseAuth authInstance;
```
Comment (biancasap) [inline src/main/java/com/google/sps/servlets/AddTA.java:25]: This should be a private FirebaseAuth authInstance.

File: src/main/java/com/google/sps/servlets/AddTA.java [hunk 3]
```diff
@@ -30,30 +35,39 @@ public void init(ServletConfig config) throws ServletException {
...
       String classCode = request.getParameter("classCode").trim();
+      Key classKey = KeyFactory.stringToKey(classCode);
+      Entity classEntity = datastore.get(classKey);
 
-      // Create a TA entity with both a user ID and a class ID
-      Entity taEntity = new Entity("TA");
-      taEntity.setProperty("userKey", userRecord.getUid());
-      taEntity.setProperty("classKey", classCode);
+      // Get the list of TAs
+      ArrayList<String> listOfClassTAs = (ArrayList) classEntity.getProperty("taList");
```
Comment (nmittu) [inline src/main/java/com/google/sps/servlets/AddTA.java:55]: This should be done in a `Transaction` that way the list won't get messed up if two people add ta's at the same time.

File: src/main/java/com/google/sps/servlets/AddTA.java [hunk 4]
```diff
@@ -30,30 +37,63 @@ public void init(ServletConfig config) throws ServletException {
...
-  // Add a TA to the datastore
+  // Add a TA to the class datastore
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
+    int retries = 10;
+
     try {
```
Comment (rjb2150) [inline src/main/java/com/google/sps/servlets/AddTA.java:48]: try { while(true) { try { ... }}} is a confusing construction -- i'd either simplify or restructure with some helper methods

**Pull request merged**